Karma/Controllers/AdminController.cs

```diff
@@ -338,11 +339,122 @@ public async Task<IActionResult> ReportReview(int reportId, string confirm, stri
             return SwitchToTabs(Enums.AdminTab.ReportReview);
         }
 
+        [HttpDelete]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> ReportReview(int reportId)
+        {
+
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToPage("/Account/Login", new { area = "Identity" });
+            }
+            
+            var oldReport = await _context.Report.FirstOrDefaultAsync(c => c.Id == reportId);
+
+            _context.Remove(oldReport);
+            await _context.SaveChangesAsync();
+
+            return SwitchToTabs(Enums.AdminTab.ReportReview);
+        }
+
+
         private void hideReportedPost(int id)
         {
             var post = _context.Post.FirstOrDefault(c => c.Id == id);
             post.State = (int)Post.PostState.Hidden;
             post.IsVisible = false;
+
+            var allSimilarReports = _context.Report.Where(a => a.PostId == id && a.ReportState == ReportStates.Open).ToListAsync();
+            allSimilarReports.Result.ForEach(a => a.ReportState = ReportStates.Approved);
+            _context.Report.UpdateRange(allSimilarReports.Result);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> ViewReports(int? page, string sorting, string? aggr)
+        {
+            const int ReportsPerPage = 6;
+            ReportDataModel reportsDM = GetFilteredReports(page, sorting, aggr, ReportsPerPage, ViewData);
+            return View(reportsDM);
+        }
+
+        private ReportDataModel GetFilteredReports(int? page, string sorting, string aggr, int reportsPerPage, ViewDataDictionary viewData)
+        {
+            var reports = _context.Report.ToList();
+            var posts = _context.Post.ToList();
+            bool filter = false;
+
+            ReportStates rptState = ReportStates.Open;
+
+            if (sorting != null)
+            {
+                filter = true;
+                viewData["filter"] = true;
+                rptState = (ReportStates)Enum.Parse(typeof(ReportStates), sorting);
+            }
+
+            var reportsDM = new ReportDataModel
+            {
+                page = page,
+                sorting = sorting,
+                aggr = aggr
+            };
+
+
+            if (aggr == "on")
+                reportsDM.Reports = GetAggregatedReports(reports, posts, reportsPerPage, rptState, filter, page, viewData);
+            else
+                reportsDM.Reports = GetPagedReports(reports, posts, reportsPerPage, rptState, filter, page, viewData);
```

Comment:
See ref1. You've used curly braces even for one line, but here you are avoiding it here. Some consistency would be nice. In addition, it is easier to edit the code and avoid mistakes in those blocks if somebody writes additional lines after if/else.

Reply:
The reason why I removed curly braces was that the function didn't have much code, but was quite long. And yeah, I agree, consistency would be nice. So I'll add curly braces.